core/frontend/src/extension/ExtensionAdmin.ts

```diff
@@ -50,60 +41,77 @@ export class ExtensionAdmin {
+   * @param provider
+   * @alpha
    */
-  public addExtensionLoaderFront(extensionLoader: ExtensionLoader) {
-    this._extensionLoaders.unshift(extensionLoader);
+  public async addExtension(provider: ExtensionProvider): Promise<void> {
+    if (provider.hostname) {
+      const hostName = provider.hostname;
+      if (this._hosts.length > 0 && this._hosts.indexOf(hostName) < 0) {
+        throw new Error(`Error loading extension: ${hostName} was not registered.`);
+      }
+    }
+    try {
+      const manifest = await provider.getManifest();
+      this._extensions.set(manifest.name, {
+        manifest,
+        execute: provider.execute,
+      });
+      // TODO - temporary fix to execute the missed startup event
+      if (manifest.activationEvents.includes("onStartup"))
+        void provider.execute();
+    } catch (e) {
+      throw new Error(`Failed to get manifest from extension: ${e}`);
+    }
   }
 
-  /** Add an ExtensionLoader to the list of extension loaders in use.
-   * @param extensionLoader Extension loader to add
+  /**
+   * Adds a list of extensions
+   * @param providers
+   * @alpha
    */
-  public addExtensionLoader(extensionLoader: ExtensionLoader) {
-    this._extensionLoaders.push(extensionLoader);
+  public async addExtensions(providers: ExtensionProvider[]): Promise<void[]> {
+    return Promise.all(
+      providers.map(async (provider) => this.addExtension(provider))
+    );
   }
 
-  /** Add an Extension intend to be bundled during compilation.
-   * @param manifestLoader A function that loads the manifest file.
-   * @param mainFunc The main function to be executed upon
+  /**
+   * Register a url (hostname) for extension hosting (i.e. https://localhost:3000, https://www.yourdomain.com, etc.)
+   * @param hostUrl
    */
-  public async addBuildExtension(manifestPromise: Promise<BuildExtensionManifest>, mainFunc?: ResolveFunc): Promise<void> {
-    const manifest = await this.getManifest(manifestPromise);
-    this._installedExtensions.set(manifest.name, { manifest, mainFunc });
+  public registerHost(hostUrl: string) {
+    const url = new URL(hostUrl).hostname.replace("www", "");
+    if (this._hosts.indexOf(url) < 0) {
+      this._hosts.push(url);
+    }
   }
 
   /** Loops over all enabled Extensions and triggers each one if the provided event is defined. */
-  private async activateExtensionEvents(event: ActivationEvent) {
-    for (const extension of this._installedExtensions.values()) {
-      if (!extension.manifest.activationEvents)
-        continue;
+  private async activateExtensionEvents(event: string) {
+    for (const extension of this._extensions.values()) {
+      if (!extension.manifest.activationEvents) continue;
       for (const activationEvent of extension.manifest.activationEvents) {
         if (activationEvent === event) {
-          this.execute(extension); // eslint-disable-line @typescript-eslint/no-floating-promises
+          void this._execute(extension);
```

Comment:
We decided to not use `void` to avoid using the eslint-disable-line and instead keep that warning in there. I've seen this in a few cases, mind switching them back?

Reply:
are you suggesting to re-add eslint-disable or just allow the warnings to occur? I don't love the void hack either, but I dont know that we should just ignore the warnings?

Reply:
re-add the eslint-disable.